processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessingContextImpl.java

```diff
@@ -90,18 +95,37 @@ public DeferredCompletion deferCompletion() {
             return CompletableFuture.completedFuture(null);
         }
 
-        CompletableFuture<Void> future = new CompletableFuture<>();
-        DeferredCompletion nextCompletion = () -> future.complete(null);
-
         DecatonTask<P> task = new DecatonTask<>(
                 this.task.metadata(), taskData, this.task.taskDataBytes());
         DecatonProcessor<P> nextProcessor = downstreams.get(0);
+        final TraceHandle parentTrace = request.trace();
+        final TraceHandle traceHandle = null == parentTrace ? NoopTrace.INSTANCE
+                                                            : parentTrace.childFor(nextProcessor);
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        DeferredCompletion nextCompletion = () -> {
+            future.complete(null);
+            try {
+                traceHandle.processingCompletion();
```

Comment:
Please add a test case in `ProcessingContextImpl` for those trace methods invocations.